src/main/java/brave/opentracing/BraveActiveSpanSource.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.opentracing;
+
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Manages the active Brave span.
+ *
+ * Note that it is important to use the OpenTracing API exclusively to activate spans. If you fail
+ * to do this, querying for the active span using the OpenTracing API will throw an exception.
+ */
+final class BraveActiveSpanSource implements ActiveSpanSource {
+  private final Map<Long, BraveActiveSpan> activeSpans = new ConcurrentHashMap<>();
```

Comment:
Will document. It's not in deference to the contract, but rather to the fact that Brave doesn't expose a way to get the current SpanInScope object. It's only available through calls to `withSpanInScope`, but you can't call that again because it will put another instance of the span on the stack, and we won't have an opportunity to close said span.

A change to Brave that exposed the current SpanInScope would greatly simplify this code. If you don't object, I can make that change as well and then revisit here.